CreateProcess error=206, The filename or extension is too long.

<!--
> Please follow the issue template below for bug reports and queries.
> Bug reports or queries opened without any of these info will be **closed** without any explanation.
> For feature requests, label the title with [FEATURE] and explain your use case and ideas clearly below, you can remove sections which are not relevant.
> For bug reports your `.yo-rc.json` file is mandatory else we will close the issue.
> You can run `yo jhipster:info` in your project folder to get most of the required info
-->
##### **Overview of the issue**

<!-- explain the issue or feature request, if an error is being thrown a stack trace helps -->

Gradle `bootRun` task of generated microservice gateway applicaton fails on Windows due to the limit of [Windows’ CreateProcess](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682425%28v=vs.85%29.aspx) function. Classpath length of default gateway application exceeds that limit if too much framework selected.

For more info:
http://tuhrig.de/gradles-bootrun-and-windows-command-length-limit/

`gradlew --stacktrace` output:
``` text
...
:bootRun FAILED

FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':bootRun'.

> A problem occurred starting process 'command 'C:\Program Files\Java\jdk1.8.0_102\bin\java.exe''
> ..
> Exception is:
> org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':bootRun'.
> ..
> Caused by: java.io.IOException: CreateProcess error=206, The filename or extension is too long
>         ... 6 more

BUILD FAILED
```
##### **Motivation for or Use Case**

<!-- explain why this is a bug for you -->

Generated microservice gateway application cannot be run on Windows.
##### **Reproduce the error**

<!-- an unambiguous set of steps to reproduce the error. If you have a JavaScript error, maybe you can provide a live example with
  [JSFiddle](http://jsfiddle.net/)? -->

Generate a microservice gateway application on Windows with preferences of following `.yo-rc.json` file
##### **Related issues**

<!-- has a similar issue been reported before? -->

`_gatling.gradle` has this issue solved #1514 
##### **Suggest a Fix**

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

Add `pathingJar` task to `_build.gradle` and make `bootRun` dependent to it

http://tuhrig.de/gradles-bootrun-and-windows-command-length-limit/
##### **JHipster Version(s)**

<!-- to provide all information we need, you can use: yo jhipster:info -->

<!-- which version of Jhipster are you using, is it a regression? -->

3.9.1/master
##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<!-- This is mandatory for bug reports. This will help us to replicate the scenario, you can remove the rememberMe key. -->

``` yaml
{
  "generator-jhipster": {
    "jhipsterVersion": "3.9.1",
    "baseName": "gateway",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "hibernateCache": "no",
    "clusteredHttpSession": false,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "gradle",
    "enableSocialSignIn": false,
    "jwtSecretKey": "",
    "useSass": false,
    "applicationType": "gateway",
    "testFrameworks": [
      "gatling",
      "cucumber",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": false
  }
}
```
##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!-- - if the error is during an entity creation or associated with a specific entity. If you are using JDL share that as well -->

There is no entity, only default gateway application
##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

Windows only

--- a/generators/server/templates/_build.gradle
+++ b/generators/server/templates/_build.gradle
@@ -1,3 +1,5 @@
+import org.gradle.internal.os.OperatingSystem
+
 buildscript {
     repositories {
         mavenLocal()
@@ -59,8 +61,31 @@ springBoot {
     buildInfo()
 }
 
-bootRun {
-    addResources = false
+if (OperatingSystem.current().isWindows()) {
+    task pathingJar(type: Jar) {
+        dependsOn configurations.runtime
+        appendix = 'pathing'
+
+        doFirst {
+            manifest {
+                attributes 'Class-Path': configurations.runtime.files.collect {
+                    it.toURL().toString().replaceFirst(/file:\/+/, '/')
+                }.join(' ')
+            }
+        }
+    }
+
+    bootRun {
+        addResources = false
+        dependsOn pathingJar
+        doFirst {
+            classpath = files("$buildDir/classes/main", "$buildDir/resources/main", pathingJar.archivePath)
+        }
+    }
+} else {
+    bootRun {
+        addResources = false
+    }
 }
 
 test {

